product_variant_default_code/models/product.py

```diff
@@ -211,43 +213,21 @@ class ProductAttribute(models.Model):
         ("number_uniq", "unique(name)", _("Attribute Name must be unique!"))
     ]
 
-    def write(self, vals):
-        if "code" not in vals:
-            return super(ProductAttribute, self).write(vals)
-        result = super(ProductAttribute, self).write(vals)
-        # Rewrite reference on all product variants affected
+    @api.onchange("code", "name", "sequence")
+    def onchange_default_code_attribute(self):
         for product in self.mapped(
             "attribute_line_ids.product_tmpl_id.product_variant_ids"
-        ).filtered(lambda x: x.product_tmpl_id.reference_mask and not x.manual_code):
-            render_default_code(product, product.reference_mask)
-        return result
+        ):
+            product.product_tmpl_id._compute_reference_mask()
 
 
 class ProductAttributeValue(models.Model):
     _inherit = "product.attribute.value"
 
-    @api.onchange("name")
-    def onchange_name(self):
-        """Set a default code for the value"""
+    @api.onchange("code", "name")
+    def onchange_default_code_values(self):
```

Comment:
Why did you rename this onchange (normally we name in onchange_name because the field that have been changed is "name")